packages/muelu/src/Smoothers/MueLu_Ifpack2Smoother_def.hpp

```diff
@@ -586,6 +586,12 @@ void Ifpack2Smoother<Scalar, LocalOrdinal, GlobalOrdinal, Node>::SetupLineSmooth
       myparamList.set("partitioner: map", TVertLineIdSmoo);
       myparamList.set("partitioner: local parts", maxPart + 1);
     } else {
+      if (myparamList.isParameter("partitioner: block size") &&
+          myparamList.get<int>("partitioner: block size") != -1) {
+        int block_size = myparamList.get<int>("partitioner: block size");
+        numLocalRows /= block_size;
```

Comment:
Since ``block_size`` comes directly from the user, here the code should probably check that ``numLocalRows`` is divisible by it.